framework/src/base/BlockRestrictable.C

```diff
@@ -27,6 +27,8 @@ InputParameters validParams<BlockRestrictable>()
   // Add the user-facing 'block' input parameter
   params.addParam<std::vector<SubdomainName> >("block", "The list of block ids (SubdomainID) that this object will be applied");
 
+  params.addPrivateParam<bool>("delay_initialization", false);
```

Comment:
Yes: I don't want the private ones to show up there... I was just pointing out that there is currently no way for a developer to ever know that private parameters exist and that they can manipulate them.

I really think in this particular case that some documentation on the Wiki is warranted that explains what the option is and why you would use it.  But... well... we have lots of things that are undocumented :-)